leaf-ui/rappid-extensions/ElementInspector.js

```diff
@@ -933,35 +1011,41 @@ var ElementInspector = Backbone.View.extend({
         }
 
         this.funcTypeChanged(null);
+        
     },
 
     /**
      * Makes corresponding changes for the cell attributes, according to the values in the
      * inspector. This function is always called alongside with updateChart
      * and updateChartUserDefined.
      */
-    updateCell: function(event) {        
+    updateCell: function(event) {   
+            
         IntentionColoring.refresh();
         changeFont(current_font, paper);
-        var funcType = this.intention.dynamicFunction.stringDynVis;
-        var initSatVal = this.intention.getInitialSatValue();
+        if (this.model.get('intention').get('evolvingFunction') != null) {
```

Comment:
Can move logic around and inline variables in this function to make it much cleaner